### PR TITLE
minor cleanup of WildcardIndex

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-actor/src/main/scala/akka/util/WildcardIndex.scala [hunk 1]
```diff
@@ -20,7 +18,7 @@ private[akka] final case class WildcardIndex[T](wildcardTree: WildcardTree[T] =
   def find(elems: Iterable[String]): Option[T] =
     (if (wildcardTree.isEmpty) {
       if (doubleWildcardTree.isEmpty) {
-        empty
+        WildcardTree[T]() // empty
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/WildcardIndex.scala]:
```
object WildcardTree { 
  private[this] val _empty = WildcardTree[Any]()
  def empty[T] = _empty.asInstanceOf[WildcardTree[T]] 
}
```

?

File: akka-actor/src/main/scala/akka/util/WildcardIndex.scala [hunk 2]
```diff
@@ -42,7 +42,7 @@ private[akka] object WildcardTree {
 
 private[akka] final case class WildcardTree[T](data: Option[T] = None, children: Map[String, WildcardTree[T]] = HashMap[String, WildcardTree[T]]()) {
 
-  lazy val isEmpty: Boolean = data.isEmpty && children.isEmpty
+  def isEmpty: Boolean = data.isEmpty && children.isEmpty
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/WildcardIndex.scala]: good 👍

**Comment by ktoso** [comment]
LGTM

File: akka-actor/src/main/scala/akka/util/WildcardIndex.scala (hunk 1 above)
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/WildcardIndex.scala]: well, that is what is there already, but named `apply`

Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/WildcardIndex.scala]: Oh, didn't see it in the PR, then all good 👍

**Comment by drewhk** [comment]
LGTM

**Comment by johanandren** [comment]
LGTM

**Pull request merged**